src/utils/misc.ts

```diff
@@ -126,3 +129,43 @@ export function getUnipool(address: Address): Unipool {
 
   return unipool;
 }
+
+export function saveTokenAllocation(
+  recipient: string,
+  txHash: string,
+  logIndex: BigInt,
+  amount: BigInt,
+  timestamp: BigInt,
+  tokenDistroAddress: string,
+): void {
+  let transactionTokenAllocations = TransactionTokenAllocation.load(txHash);
+  if (!transactionTokenAllocations) {
+    transactionTokenAllocations = new TransactionTokenAllocation(txHash);
+  }
+  const tokenAllocationIds =
+    transactionTokenAllocations.tokenAllocationIds || [];
+  const entityId = `${txHash}-${logIndex}`;
+  const entity = new TokenAllocation(entityId);
+  entity.amount = amount;
+  entity.timestamp = timestamp;
+  entity.recipient = recipient;
+  entity.txHash = txHash;
+  entity.tokenDistroAddress = tokenDistroAddress;
+  entity.save();
+  tokenAllocationIds.push(entityId);
+  transactionTokenAllocations.tokenAllocationIds = tokenAllocationIds;
+  transactionTokenAllocations.save();
+}
+
+export function getTokenDistroBalance(
+  tokenDistro: Address,
+): TokenDistroBalance {
```

Comment:
Why is it changed again? TokenDistroBalance id is combination of tokenDistro address and user address! on entity is saved for each tokendisto-user pair